server/raft.go

```diff
@@ -1069,7 +1069,18 @@ func (n *raft) shutdown(shouldDelete bool) {
 		return
 	}
 	close(n.quit)
-	n.c.closeConnection(InternalClient)
+	if c := n.c; c != nil {
+		var subs []*subscription
+		c.mu.Lock()
+		for _, sub := range c.subs {
+			subs = append(subs, sub)
+		}
+		c.mu.Unlock()
+		for _, sub := range subs {
+			n.unsubscribe(sub)
+		}
+		c.closeConnection(InternalClient)
```

Comment:
Note that unsubscribe and closing a client will end-up grabbing the server lock (or others due to update of some objects, route maps, etc..), so it may need to be done outside of n.Lock() *if* you have situations where you have server lock -> raft lock.

Reply:
Will take a look, and want to get a green test run so will look at the flappers in a bit.